crates/api/src/instance.rs

```diff
@@ -192,14 +187,13 @@ impl Instance {
     /// the value, if found.
     ///
     /// Returns `None` if there was no export named `name`.
-    pub fn get_export(&self, name: &str) -> Option<&Extern> {
-        let (i, _) = self
-            .module
```

Comment:
This may need double-checking, but I think `self.module` may not be necessary any more after this change. I think the `Store` is still needed but we may be able to remove the `Module` field from an `Instance`?

Reply:
I did a quick experiment with this, and `Linker::instance` currently needs `Instance::module`. We don't currently store the export names in the instance otherwise.

Reply:
Ah ok no worries. We could make the internal handle `pub(crate)` so the linker could access it, but it's fine to tackle this later if really necessary.

Reply:
And, it turns out, we can actually make `Instance::exports` provide the names in a reasonable way, which helps the `Linker`, and makes the API much nicer as you no longer need to `zip` with `Module::exports` just to get the names.

Now, nothing in `Instance` itself or its users need it to have a `Module`. I've left it in for now because it looks like it might unregister some things in its `Drop`, but if we want to sort that out we could drop the `Module` after instantiation.